subsys/net/ip/tcp2.c

```diff
@@ -965,6 +965,7 @@ int net_tcp_update_recv_wnd(struct net_context *context, s32_t delta)
 /* net context wants to queue data for the TCP connection */
 int net_tcp_queue_data(struct net_context *context, struct net_pkt *pkt)
 {
+	int irq_level = irq_lock();
```

Comment:
What do we want to protect here, can you add more explanation to commit msg? Locking irqs seems quite heavy thing to do.

Reply:
@jukkar 
> can you add more explanation to commit msg? Locking irqs seems quite heavy thing to do.

The message about interrupt proteciton is refined, please take a look.

Reply:
Will net_tcp_queue_data() actually be called from the interrupt context, or is the lock protecting simultaneous access to this function or the context?

Reply:
Why mutex or semaphore is not enough here?

Reply:
@jukkar @pfl 

You are right, a mutex is a better idea here, so droped irq locks and added mutex, please take a look.